packages/core/src/system.ts

```diff
@@ -125,7 +131,18 @@ export function createSystem<T extends ActorSystemInfo>(
         delete timerMap[scheduledEventId];
         delete system._snapshot._scheduledEvents[scheduledEventId];
 
-        system._relay(source, target, event);
+        const resolvedTarget =
+          typeof target === 'string'
+            ? (source.getSnapshot() as AnyMachineSnapshot).children[target]
+            : target;
+
+        if (!resolvedTarget) {
+          throw new Error(
+            `Actor with id ${typeof target === 'string' ? target : target.sessionId} not found in the system.`
+          );
+        }
```

Comment:
IMHO this doesn't fit here. It's forcefully trying to make something pass without considering what area of the code was designed to deal with this kind of resolving. `Scheduler` works with actor refs so even assuming here that the source is a machine actor is already a big stretch. The target should always come pre-resolved to the Scheduler as it was before.